app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,46 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = current_user
     @posts = @user.recent_posts.includes(:comments)
   end
 
   def show
     @post = Post.includes(:comments).find(params[:id])
+    @user = @post.author
+    @comment = Comment.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to user_posts_path(current_user), notice: 'Post was successfully created.'
+    else
+      @posts = current_user.recent_posts.includes(:comments)
+      render 'new'
+    end
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    existing_like = @post.likes.find_by(user: current_user)
+
+    if existing_like
+      existing_like.destroy
+      @post.decrement!(:likes_counter)
+    else
+      current_user.likes.create(post: @post)
+    end
+
+    redirect_to user_post_path(@post.author, @post)
+  end
+
+  def new
```

Comment:
Good job so far but I think you should have create in the **views/posts/new.html.erb** file which accepts input to make user to be able to create posts. Please fix them 👍

Reply:
I have this one called `_new.html.erb` located in the `./app/views/users`. @codecaiine  I am not sure what should I add here, could you please clarify more?